### ISSUE TITLE
What css/js libraries/frameworks do you want included in kodeWeave?

### ISSUE THREAD

**Issue by mikethedj4** [mikethedj4/kodeWeave#4]
What CSS/JavaScript Libraries/Frameworks would you like to be included with kodeWeave?

**Comment by steakknife** [comment]
Perhaps *browser-side* packages >= 90% per http://www.javascripting.com/ and/or top 72 from https://www.npmjs.com/browse/star

Long short-list:

- Angular 1/2
- Ember
- Vue
- Backbone / Underscore
- Three.js
- d3
- enhancejs 
- modernizr
- handlebars
- immutable
- lodash
- moment
- moment-timezone
- Angular Material
- Bootstrap
- Foundation 5/6
- (jQuery family)
- Polymer (experimental)
- (react family)
- Semantic UI
- CoffeeScript
- Babel (ES6, etc.)
- Elm
- TypeScript

HTH

**Comment by AlexandreBonneau** [comment]
Well, by default you could add those, but there are so many you should probably stick to 5 or 6 pre-defined, and add inputs for the user to use the links they want to (and allow those inputs to be reordered, ala codepen).

For preprocessors, I would say SCSS is a must-have, and LESS too for those of us who still haven't see the SCSS light!

Babel is also mandatory if you want to use ES6, and Typescript too.

Well, in fact, just take a look at which option codepen is using and that should be a good base ;)

**Comment by mikethedj4** [comment]
I apologize for the late reply. I'd love to get these integrated but easier said than done. 

Remember with libraries when the user downloads their weave as a zip the goal is to have all the checked libraries downloaded as well. The reason why I never added the option for users to add in their own libraries is Due to the [Same-origin policy](https://en.m.wikipedia.org/wiki/Same-origin_policy) many libraries that are added dynamically like that won't be able to be downloaded/compacted into the executed zip file.

**Comment by stevenventimiglia** [comment]
I agree with @AlexandreBonneau about Sass (SCSS).  It's become more of a requirement and less of an option over the past couple of years.

**Comment by starlocke** [comment]
1. I'm still an "old-school" LESS guy.
2. jQuery 3 introduces breaking changes, so it's also nice to have [jquery-migrate](https://github.com/jquery/jquery-migrate) around to smooth out issues for people who are "upgrading" their codebase but haven't/cannot squash out all the jQuery 1 and 2 stuff.

**Comment by mikethedj4** [comment]
I'm slowly but surely adding these in. I just added LESS in today. along with jquery migrate, I've added hint.css, animate.css, material design lite, Jarallax, and more. I recently came across [sass.js](https://github.com/medialize/sass.js); however have been having difficulty [getting it to run](https://stackoverflow.com/questions/45178906/running-sass-js-in-the-browser). Once I figure that out I will definitely be adding that. along with some other goodies.
